packages/@aws-cdk/custom-resources/lib/provider-framework/runtime/framework.ts

```diff
@@ -112,17 +115,38 @@ async function invokeUserFunction<A extends { ResponseURL: '...' }>(functionArnE
 
   log('user function response:', resp, typeof(resp));
 
+  // parse function name from arn
+  // arn:${Partition}:lambda:${Region}:${Account}:function:${FunctionName}
+  const arn = functionArn.split(':');
+  const functionName = arn[arn.length - 1];
+
   const jsonPayload = parseJsonPayload(resp.Payload);
   if (resp.FunctionError) {
+    let totalSleep = 0, attempt = 0;
+    while (totalSleep <= MAX_TOTAL_SLEEP) {
+      // if the user's lambda has become Inactive, we must retry the invocation until Lambda finishes provisioning resources for it.
+      const getFunctionResponse = await getFunction({
+        FunctionName: functionName,
```

Comment:
Looks like we can just provide the ARN here instead of trying to parse per [GetFunction Documentation](https://docs.aws.amazon.com/lambda/latest/dg/API_GetFunction.html).

Reply:
We can, but we need the name later anyway, and imo passing the name is cleaner. If you prefer though we can change this to the ARN